test/unit/high-security.test.js

```diff
@@ -41,238 +35,278 @@ function getPath(obj, path) {
+        check(t, 'attributes.include', [], ['foobar'])
+        check(t, 'attributes.exclude', [], ['fizzbang', 'request.parameters.*'])
+        t.end()
       })
 
-      it('should not change transaction_tracer settings', function () {
-        check('transaction_tracer.record_sql', 'obfuscated', 'raw')
-        check('transaction_tracer.attributes.include', [], ['foobar'])
-        check('transaction_tracer.attributes.exclude', [], ['fizzbang'])
+      t.test('should not change transaction_tracer settings', function (t) {
+        check(t, 'transaction_tracer.record_sql', 'obfuscated', 'raw')
+        check(t, 'transaction_tracer.attributes.include', [], ['foobar'])
+        check(t, 'transaction_tracer.attributes.exclude', [], ['fizzbang'])
+        t.end()
       })
 
-      it('should not change error_collector settings', function () {
-        check('error_collector.attributes.include', [], ['foobar'])
-        check('error_collector.attributes.exclude', [], ['fizzbang'])
+      t.test('should not change error_collector settings', function (t) {
+        check(t, 'error_collector.attributes.include', [], ['foobar'])
+        check(t, 'error_collector.attributes.exclude', [], ['fizzbang'])
+        t.end()
       })
 
-      it('should not change browser_monitoring settings', function () {
-        check('browser_monitoring.attributes.include', [], ['foobar'])
-        check('browser_monitoring.attributes.exclude', [], ['fizzbang'])
+      t.test('should not change browser_monitoring settings', function (t) {
+        check(t, 'browser_monitoring.attributes.include', [], ['foobar'])
+        check(t, 'browser_monitoring.attributes.exclude', [], ['fizzbang'])
+        t.end()
       })
 
-      it('should not change transaction_events settings', function () {
-        check('transaction_events.attributes.include', [], ['foobar'])
-        check('transaction_events.attributes.exclude', [], ['fizzbang'])
+      t.test('should not change transaction_events settings', function (t) {
+        check(t, 'transaction_events.attributes.include', [], ['foobar'])
+        check(t, 'transaction_events.attributes.exclude', [], ['fizzbang'])
+        t.end()
       })
 
-      it('should shut down the agent if high_security is false', function () {
+      t.test('should shut down the agent if high_security is false', function (t) {
         config.onConnect({ high_security: false })
-        expect(config.agent_enabled).to.be.false
+        t.equal(config.agent_enabled, false)
+        t.end()
       })
 
-      it('should shut down the agent if high_security is missing', function () {
+      t.test('should shut down the agent if high_security is missing', function (t) {
         config.onConnect({})
-        expect(config.agent_enabled).to.be.false
+        t.equal(config.agent_enabled, false)
+        t.end()
       })
 
-      function check(key, expected, server) {
+      function check(t, key, expected, server) {
```

Comment:
it might be better to extend tap to add this as an assertion so you can just call `t.check`

See: https://github.com/newrelic/node-newrelic/pull/868/files#diff-47318341a886cbc8f43b379aa814ec406a831116ea1540b77de2f4caa4f41cefR14

Reply:
Great, I've added the custom asserts. Thanks for the tip :)